defaults/main.yml

```diff
@@ -116,6 +116,16 @@ checkmk_agent__user_home: '/var/lib/nagios'
 # SSH Public key for logging in with SSH user.
 checkmk_agent__user_key: ''
 
+
+# .. envvar:: checkmk_agent__authorized_keys__dependent_list
+#
+# Authorized key configuration for the ``debops.authorized_keys`` role.
+checkmk_agent__authorized_keys__dependent_list:
+  - name: '{{ checkmk_agent__ssh_user }}'
+    ssh_keys: [ '{{ checkmk_agent__user_key }} if {{ checkmk_agent__user_key|d() }} else {{ hostvars[groups["debops_service_checkmk_server"][0]].ansible_local.checkmk_server.ssh_publickey|d("") }}' ]
```

Comment:
For this you will probably want to split the role into normal and "env" role (from "environment"). It would work like this:
```
- role: `checkmk_server/env'
- role: `debops.authorized_keys'
- role: `checkmk_server`

That way you can prepare data for other roles that need it, hand it off to the other roles and return back to managing your application. For an example, check the [debops.php](https://github.com/debops/ansible-php) role and its example playbook.